src/cmd/promremotebench/write_test.go

```diff
@@ -40,83 +38,72 @@ import (
-		numHosts int
-		expectedBatches int
-		expectedSeries int
-	} {
-		{
-			name: "one host",
-			numHosts: 1,
-		},
-		{
-			name: "eleven hosts",
-			numHosts: 11,
-		},
-		{
-			name: "hundred hosts",
-			numHosts: 100,
-		},
-	}
+	numBatchesRecieved := 0
+	numTSRecieved := 0
+	var wg sync.WaitGroup
 
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			numBatchesRecieved := 0
-			numTSRecieved := 0
-			var wg sync.WaitGroup
-		
-			server := httptest.NewServer(
-				http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-					compressed, err := ioutil.ReadAll(r.Body)
-					if err != nil {
-						http.Error(w, err.Error(), http.StatusInternalServerError)
-						return
-					}
-		
-					reqBuf, err := snappy.Decode(nil, compressed)
-					if err != nil {
-						http.Error(w, err.Error(), http.StatusBadRequest)
-						return
-					}
-		
-					var req prompb.WriteRequest
-					if err := proto.Unmarshal(reqBuf, &req); err != nil {
-						http.Error(w, err.Error(), http.StatusBadRequest)
-						return
-					}
-		
-					numBatchesRecieved++
-					numTSRecieved += len(req.Timeseries)
-		
-					wg.Done()
-				}),
-			)
-		
-			serverURL, err := url.Parse(server.URL)
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			compressed, err := ioutil.ReadAll(r.Body)
 			if err != nil {
-				t.Fatal(err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
 			}
-		
-			remotePromClient, err := NewClient(serverURL.String(), time.Minute)
+
+			reqBuf, err := snappy.Decode(nil, compressed)
 			if err != nil {
-				t.Fatal(err)
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
 			}
-		
-			hostGen := generators.NewHostsSimulator(test.numHosts, time.Now(),
-				generators.HostsSimulatorOptions{})
-			series := hostGen.Generate(0)
 
-			batchSize := 10
-			expectedBatches := int(math.Ceil(float64(len(series))/float64(batchSize)))
+			var req prompb.WriteRequest
+			if err := proto.Unmarshal(reqBuf, &req); err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
 
-			wg.Add(expectedBatches)
-			remoteWrite(series, remotePromClient, batchSize)
-			wg.Wait()
-			assert.Equal(t, expectedBatches, numBatchesRecieved)
-			assert.Equal(t, len(series), numTSRecieved)
+			numBatchesRecieved++
+			numTSRecieved += len(req.Timeseries)
 
-			fmt.Println("wrote series:", len(series))
-		})
+			wg.Done()
+		}),
+	)
+
+	serverURL, err := url.Parse(server.URL)
+	if err != nil {
+		t.Fatal(err)
 	}
+
+	remotePromClient, err := NewClient(serverURL.String(), time.Minute)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	hostGen := generators.NewHostsSimulator(1, 10, time.Now(), generators.HostsSimulatorOptions{})
+	series := hostGen.Generate(0, 0)
+	wg.Add(11)
+	remoteWrite(series, remotePromClient, 10)
+	wg.Wait()
+	assert.Equal(t, 11, numBatchesRecieved)
+	assert.Equal(t, 101, numTSRecieved)
+
+	numBatchesRecieved = 0
+	numTSRecieved = 0
```

Comment:
Instead of resetting, if these were multiple tests you wouldn't have manual reset/cleanup steps before running another workload.